### PR TITLE
Safe leaderboard avatar loading

### PR DESCRIPTION
**Pull request by zachstence**
### Description

As requested by @Miodec here
https://github.com/monkeytypegame/monkeytype/pull/3162#issuecomment-1160331780

- Refactor the Discord avatar loading logic I wrote in #3162  into `loadDiscordAvatar` util function
- Use `loadDiscordAvatar` in both the account button and leaderboard avatars
- Adjust leaderboard avatar styling

leaderboard avatars

### CODE REVIEW

File: frontend/src/ts/elements/leaderboards.ts [hunk 1]
```diff
@@ -276,23 +277,16 @@ function fillTable(lb: LbKey, prepend?: number): void {
...
-
-    const isCurrentUser =
-      Auth.currentUser &&
-      entry.uid === Auth.currentUser.uid &&
-      snap.discordAvatar &&
-      snap.discordId;
-
-    const entryHasAvatar = entry.discordAvatar && entry.discordId;
-
-    const avatarSource = (isCurrentUser && snap) || (entryHasAvatar && entry);
```
Comment (Bruception) [inline frontend/src/ts/elements/leaderboards.ts:291]: This logic determined where to source the avatar data from.

File: frontend/src/ts/elements/account-button.ts [hunk 2]
```diff
@@ -23,27 +24,21 @@ export function loading(truefalse: boolean): void {
...
   if (Auth.currentUser != null) {
-    if (discordAvatar && discordId) {
-      // Replace font-awesome account icon with Discord avatar only if it loads successfully
-      // https://stackoverflow.com/a/5058336/9080819
-      const avatarUrl = `https://cdn.discordapp.com/avatars/${discordId}/${discordAvatar}.png`;
-      $("<img/>")
-        .attr("src", avatarUrl)
-        .on("load", (event) => {
-          $(event.currentTarget).remove();
+    loadDiscordAvatar(discordId, discordAvatar)
```
Comment (Bruception) [inline frontend/src/ts/elements/account-button.ts:27]: Use async/await over promises

File: frontend/src/ts/utils/load-discord-avatar.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    const avatarUrl = `https://cdn.discordapp.com/avatars/${discordId}/${discordAvatar}`;
+
+    $("<img/>")
+      .attr("src", avatarUrl)
+      .on("load", (event) => {
+        $(event.currentTarget).remove();
+        resolve(avatarUrl);
+      })
+      .on("error", () => reject());
+  });
```
Comment (Miodec) [inline frontend/src/ts/utils/load-discord-avatar.ts:24]: Since this is meant to be reused - try to use vanilla for performance (it will surely be faster)

**Review by Miodec — CHANGES_REQUESTED** [review]

**Comment by Bruception** [comment]
I'll be re-working this PR. Thank you for the inspiration :)

**Pull request closed without merging**